cmd/juju/storage/volume.go

```diff
@@ -35,3 +38,62 @@ func NewVolumeSuperCommand() cmd.Command {
 type VolumeCommandBase struct {
 	StorageCommandBase
 }
+
+// VolumeInfo defines the serialization behaviour of the storage volume (currently, disk) information.
+type VolumeInfo struct {
+	Attachments map[string]AttachmentInfo
+}
+
+type AttachmentInfo struct {
+	Storage     string  `yaml:"storage" json:"storage"`
+	Assigned    bool    `yaml:"assigned" json:"assigned"`
+	Machine     string  `yaml:"machine" json:"machine"`
+	Attached    bool    `yaml:"attached" json:"attached"`
+	DeviceName  string  `yaml:"device-name" json:"device-name"`
+	Size        *uint64 `yaml:"size" json:"size"`
```

Comment:
This should be omitempty if we are making it a pointer and thus allowing it to be nil
The other string fields should also probably be omitempty for completeness, although I think only FileSystem might be ""